azkaban-common/src/main/java/azkaban/executor/RunningExecutionsUpdater.java

```diff
@@ -133,24 +135,31 @@ public void updateExecutions() {
   }
 
   private void handleException(final Entry<Optional<Executor>, List<ExecutableFlow>> entry,
-      final Executor executor, final ExecutorManagerException e) {
+      final Executor executor, final ExecutorManagerException e,
+      final ArrayList<ExecutableFlow> finalizeFlows) {
     logger.error("Failed to get update from executor " + executor.getHost(), e);
     boolean sendUnresponsiveEmail = false;
+    final boolean executorRemoved = executorRemoved(executor.getId());
     for (final ExecutableFlow flow : entry.getValue()) {
       final Pair<ExecutionReference, ExecutableFlow> pair =
           this.runningExecutions.get().get(flow.getExecutionId());
-      // TODO can runningFlows.get ever return null, causing NPE below?
 
       this.updaterStage
           .set("Failed to get update for flow " + pair.getSecond().getExecutionId());
 
-      final ExecutionReference ref = pair.getFirst();
-      ref.setNextCheckTime(DateTime.now().getMillis() + this.errorThreshold);
-      ref.setNumErrors(ref.getNumErrors() + 1);
-      if (ref.getNumErrors() == this.numErrorsBeforeUnresponsiveEmail
-          || ref.getNumErrors() % this.numErrorsBetweenUnresponsiveEmail == 0) {
-        // if any of the executions has failed many enough updates, alert
-        sendUnresponsiveEmail = true;
+      if (executorRemoved) {
+        logger.warn("Finalizing execution " + flow.getExecutionId()
+            + ". Executor id of this execution doesn't exist");
```

Comment:
The logging message here is not accurate, please update it to reflect "executor is removed".

Reply:
> The logging message here is not accurate, please update it to reflect "executor is removed".

I changed that as requested.

Although I don't see how the message `Executor id of this execution doesn't exist` is not accurate. To me it seems to say the same thing + a bit more context. To be specific, this part of code _doesn't necessarily know that executor has been removed_. All it knows is that the executor id that the execution is assigned to _doesn't exist_. From this code's point of view it could also be that such executor id never existed :) Although with current Azkaban code that shouldn't ever happen of course – it's not possible that an execution is assigned an executor id that _never existed_.

Reply:
I think we are talking about different pieces of the code. Because originally the logger messages were the same for line 84 and line 152, I suggest changing the one on line 152 to reflect the reason "executor is removed" not line 84.

Reply:
Oh, sorry, I already forgot that there are two similar cases. I think they are about the same thing, could you check please? I believe the warn message could be unified for these. Should I extract a method? What would you want the message to be?

Reply:
I think they are still slightly different. The previous one checks whether the executor is null in the running executions cache. And this one checks if the executor exists in DB. For better debugging purpose, I would suggest keeping two different logging messages.

Reply:
> For better debugging purpose, I would suggest keeping two different logging messages.

Great thinking 👍 I'll just flip them around then to match with your original request. Sorry for my mistake with the wrong line. I would still say that "doesn't exist" is more exact than "is removed" (which could be a stretch) but both work for me..

> The previous one checks whether the executor is null in the running executions cache

That can only happen immediately after startup btw, because running executions are not updated from the DB afterwards.